From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Oli Lan <olilan@google.com>
Date: Tue, 19 Jul 2022 10:45:22 +0000
Subject: [PATCH] Revert "RESTRICT AUTOMERGE Prevent non-admin users from
 deleting system apps."

This reverts commit 4005549db2fa7e1524fc0dbbe22c774fb00b6cb3.

Reason for revert: Regression, DELETE_SYSTEM_APP flag no longer works

Change-Id: I7386d1ba3d61b95836b85c52214c83b216c478e8
(cherry picked from commit 49d8f9325a8d103497632097010899f87f403faa)
Merged-In: I7386d1ba3d61b95836b85c52214c83b216c478e8
---
 .../com/android/server/pm/PackageManagerService.java   | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index e8532ce4edd3..dc44fe17722d 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -18476,16 +18476,6 @@ public class PackageManagerService extends IPackageManager.Stub
                 return PackageManager.DELETE_FAILED_INTERNAL_ERROR;
             }
 
-            if (isSystemApp(uninstalledPs)) {
-                UserInfo userInfo = sUserManager.getUserInfo(userId);
-                if (userInfo == null || !userInfo.isAdmin()) {
-                    Slog.w(TAG, "Not removing package " + packageName
-                            + " as only admin user may downgrade system apps");
-                    EventLog.writeEvent(0x534e4554, "170646036", -1, packageName);
-                    return PackageManager.DELETE_FAILED_USER_RESTRICTED;
-                }
-            }
-
             // Static shared libs can be declared by any package, so let us not
             // allow removing a package if it provides a lib others depend on.
             pkg = mPackages.get(packageName);
